integration-tests/emulated/tests/bridges/bridge-hub-polkadot/src/tests/snowbridge.rs

```diff
@@ -459,6 +460,184 @@ fn send_token_from_ethereum_to_asset_hub() {
+
+		// Register ETH
+		assert_ok!(<AssetHubPolkadot as AssetHubPolkadotPallet>::ForeignAssets::force_create(
+			RuntimeOrigin::root(),
+			ether_location.clone(),
+			ethereum_sovereign_account().into(),
+			true,
+			MIN_ETHER_BALANCE,
+		));
+
+		assert_expected_events!(
+			AssetHubPolkadot,
+			vec![
+				RuntimeEvent::ForeignAssets(pallet_assets::Event::ForceCreated { .. }) => {},
+			]
+		);
+	});
+
+	// Send ether from Bridge Hub
+	BridgeHubPolkadot::execute_with(|| {
+		type RuntimeEvent = <BridgeHubPolkadot as Chain>::RuntimeEvent;
+
+		assert_ok!(<BridgeHubPolkadot as Chain>::System::set_storage(
+			<BridgeHubPolkadot as Chain>::RuntimeOrigin::root(),
+			vec![(EthereumGatewayAddress::key().to_vec(), H160(GATEWAY_ADDRESS).encode())],
+		));
+
+		let message_id: H256 = [1; 32].into();
+		// Construct SendToken message and sent to inbound queue
+		let message = VersionedMessage::V1(MessageV1 {
+			chain_id: CHAIN_ID,
+			command: Command::SendToken {
+				token: [0; 20].into(),
+				destination: Destination::AccountId32 {
+					id: AssetHubPolkadotReceiver::get().into(),
+				},
+				amount: MIN_ETHER_BALANCE + TOKEN_AMOUNT,
+				fee: XCM_FEE,
+			},
+		});
+		// Convert the message to XCM
+		let (xcm, _) = EthereumInboundQueue::do_convert(message_id, message).unwrap();
+		// Send the XCM
+		let _ = EthereumInboundQueue::send_xcm(xcm, AssetHubPolkadot::para_id()).unwrap();
+
+		// Check that the message was sent
+		assert_expected_events!(
+			BridgeHubPolkadot,
+			vec![
+				RuntimeEvent::XcmpQueue(cumulus_pallet_xcmp_queue::Event::XcmpMessageSent { .. }) => {},
+			]
+		);
+	});
+
+	// Receive ether on Asset Hub.
+	AssetHubPolkadot::execute_with(|| {
+		type RuntimeEvent = <AssetHubPolkadot as Chain>::RuntimeEvent;
+
+		// Check that the token was received and issued as a foreign asset on AssetHub
+		assert_expected_events!(
+			AssetHubPolkadot,
+			vec![
+				RuntimeEvent::ForeignAssets(pallet_assets::Event::Issued { .. }) => {},
+			]
+		);
+	});
+
+	let treasury_account_before = BridgeHubPolkadot::execute_with(|| {
+		<<BridgeHubPolkadot as BridgeHubPolkadotPallet>::Balances as frame_support::traits::fungible::Inspect<_>>::balance(&RelayTreasuryPalletAccount::get())
+	});
+
+	// Send ether from Asset Hub.
+	AssetHubPolkadot::execute_with(|| {
+		type RuntimeOrigin = <AssetHubPolkadot as Chain>::RuntimeOrigin;
+
+		let assets = vec![Asset { id: AssetId(ether_location), fun: Fungible(TOKEN_AMOUNT) }];
+		let versioned_assets = VersionedAssets::from(Assets::from(assets));
+
+		let destination = VersionedLocation::from(Location::new(
+			2,
+			[GlobalConsensus(Ethereum { chain_id: CHAIN_ID })],
+		));
+
+		let beneficiary = VersionedLocation::from(Location::new(
+			0,
+			[AccountKey20 { network: None, key: ETHEREUM_DESTINATION_ADDRESS }],
+		));
+
+		let free_balance_before =
+			<AssetHubPolkadot as AssetHubPolkadotPallet>::Balances::free_balance(
+				AssetHubPolkadotReceiver::get(),
+			);
+		// Send the Weth back to Ethereum
```

Comment:
Should be `ether` here.

Reply:
Addressed in https://github.com/polkadot-fellows/runtimes/pull/548/commits/7d39a0a5fcd3d3764bbeb0e8a30e7ec4ba873b9d